.github/workflows/ci.yml

```diff
@@ -46,9 +74,9 @@ jobs:
             ${{ env.XDG_CACHE_HOME }}/vcpkg/archives
             ${{ env.LOCALAPPDATA }}\vcpkg\archives
             ${{ env.APPDATA }}\vcpkg\archives
-          key: ${{ runner.os }}-${{ matrix.compiler }}-${{env.BUILD_TYPE}}-${{ hashFiles('**/CMakeLists.txt') }}-${{ hashFiles('./conanfile.txt')}}-${{ hashFiles('./vcpkg.json')}}
+          key: ${{ runner.os }}-${{ matrix.compiler }}-${{matrix.build_type}}-${{matrix.developer_mode}}-${{ hashFiles('**/CMakeLists.txt') }}-${{ hashFiles('./conanfile.txt')}}-${{ hashFiles('./vcpkg.json')}}
```

Comment:
```suggestion
          key: ${{ runner.os }}-${{ matrix.compiler }}-${{matrix.build_type}}-${{matrix.generator}}-${{matrix.developer_mode}}-${{ hashFiles('**/CMakeLists.txt') }}-${{ hashFiles('./conanfile.txt')}}-${{ hashFiles('./vcpkg.json')}}
```

Reply:
Since I have no idea how caching works here, I thought adding the generator was a bad idea. Which generator is used does not affect the tooling or OS setup at all, so shouldn't it share cache with other builds that are otherwise the same?

Reply:
Also: should the cache come after setupcpp is run? that seems to take a long time.

Reply:
I agree that this should not change the binary files. I am not sure if this happens in practice. Does the MSVC generator and Ninja generate the same binaries if the compilers are the same?  What about Conan, does it manage this situation?

Reply:
I think I'm fundamentally confused on how the cache works.

Based on the logs from github, it seems the cache only caches up to the point where it is called. 

![image](https://user-images.githubusercontent.com/234279/151682651-1b97b41d-7378-4010-9db5-7780991d2e53.png)

Which is very early in the process, and doesn't really seem to be gaining us much at all.

Reply:
Indeed - it's not using a conan cache *at all* because of where it's restoring from, from what I can tell. What am I missing?

Reply:
Nevermind, I'm totally wrong here. I think I see what's happening now.